### DISCUSSION TITLE
Does payloadcms provide any migration tools?

### DISCUSSION THREAD

**Discussion by blueMountain007** [payloadcms/payload#857] [Q&A] [answered]
just like [contentful-migration](https://www.npmjs.com/package/contentful-migration).

Planning to migrate from [contentful](https://www.contentful.com/) to payloadcms

**Comment by jmikrut** [comment, marked as answer]
Hey @zhounewz — thanks for the question! 

Right now, because Payload has a very powerful Local Node API, you can write migration scripts in your code very easily. As long as you can export content from Contentful in JSON or similar, you can then loop over that content in a script, and for each doc, create a doc within Payload using the local API. It's pretty easy!

**Reply by blueMountain007** [reply]
just like below?
```
const pageReq = await fetch(`${process.env.NEXT_PUBLIC_SERVER_URL}/api/posts?limit=100`)
```

**Reply by denolfe** [reply]
@zhounewz [Here is an example](https://github.com/payloadcms/public-demo/blob/0921b1a74d27ce14dc38f3f68743b07a01bcaf57/src/cron/reset.ts) of using the [local API](https://payloadcms.com/docs/local-api/overview) to seed data. You would use this same method to read and then modify the data to your liking.

Another option is using a migration tool like [mongoose-migrate](https://github.com/balmasi/migrate-mongoose) or [migrate-mongo](https://github.com/seppevs/migrate-mongo)

**Reply by blueMountain007** [reply]
@denolfe Thank you for such a detailed answer. I just wrote a simple demo with [rest api](https://payloadcms.com/docs/rest-api/overview) and it ran successfully. so I wonder if [rest api](https://payloadcms.com/docs/rest-api/overview) is also an option?
By the way, the [contentful](https://www.contentful.com/) used before and the [payload](https://payloadcms.com/) used now are distributed in 2 different code repositories, in my opinion, rest api may be an easy solution. What do you think

**Reply by denolfe** [reply]
Yes, REST API is always an option. The reason local API is usually a better option is there is less overhead, and it circumvents access control.

**Comment by DanRibbens** [comment]
I made a sample repo that has migrations inside a Payload project using `migrate-mongo` and it worked pretty well in my testing, see https://github.com/payloadcms/migrate-mongo-example.

I'll write a blog post about it and depending on reception, it could be added to the Payload docs.
